src/main/org/deidentifier/arx/DataSource.java

```diff
@@ -1,343 +1,340 @@
+        return createCSVSource(new File(file), charset, separator, containsHeader);
+    }
+
+    /**
+     * Creates an Excel data source.
+     *
+     * @param file
+     * @param sheetIndex
+     * @param containsHeader
+     * @return
+     */
+    public static DataSource createExcelSource(File file, int sheetIndex, boolean containsHeader) {
+        return new DataSource(file, sheetIndex, containsHeader);
+    }
+
+    /**
+     * Creates an Excel data source.
+     *
+     * @param file
+     * @param sheetIndex
+     * @param containsHeader
+     * @return
+     */
+    public static DataSource createExcelSource(String file, int sheetIndex, boolean containsHeader) {
+        return createExcelSource(new File(file), sheetIndex, containsHeader);
+    }
+
+    /**
+     * Creates a JDBC data source.
+     *
+     * @param url
+     * @param table
+     * @return
+     * @throws SQLException
+     */
+    public static DataSource createJDBCSource(String url, String table) throws SQLException {
+        return new DataSource(url, table);
+    }
+
+    /**
+     * Creates a JDBC data source.
+     *
+     * @param url
+     * @param user
+     * @param password
+     * @param table
+     * @return
+     * @throws SQLException
+     */
+    public static DataSource createJDBCSource(String url, String user, String password, String dbName, String schemaName, String table, int maxRows) throws SQLException {
+        return new DataSource(url, user, password, dbName, schemaName, table, maxRows);
+    }
+
+    /**
+     * The config.
+     */
+    private final ImportConfiguration config;
+
+    /**
+     * Creates a CSV source.
+     *
+     * @param file
+     * @param separator
+     * @param containsHeader
+     */
+    private DataSource(File file, Charset charset, char separator, boolean containsHeader) {
+        config = new ImportConfigurationCSV(file.getAbsolutePath(), charset, separator, containsHeader);
+    }
+
+    /**
+     * Creates an Excel source.
+     *
+     * @param file
+     * @param sheetIndex
+     * @param containsHeader
+     */
+    private DataSource(File file, int sheetIndex, boolean containsHeader) {
+        config = new ImportConfigurationExcel(file.getAbsolutePath(), sheetIndex, containsHeader);
+    }
+
+    /**
+     * Creates a JDBC data source.
+     *
+     * @param url
+     * @param table
+     * @throws SQLException
+     */
+    private DataSource(String url, String table) throws SQLException {
+        config = new ImportConfigurationJDBC(url, table);
+    }
+
+    /**
+     * Creates a JDBC data source.
+     *
+     * @param url
+     * @param user
+     * @param password
+     * @param table
+     * @throws SQLException
+     */
+    private DataSource(String url, String user, String password, String dbName, String schemaName, String table, int maxRows) throws SQLException {
+        config = new ImportConfigurationJDBC(url, user, password, dbName, schemaName, table, maxRows);
```

Comment:
Same here. Please create a new, additional constructor. maxRows is probably not needed.